sklearn/datasets/samples_generator.py

```diff
@@ -1204,10 +1214,11 @@ def make_sparse_spd_matrix(dim=1, alpha=0.95, norm_diag=False,
     prec = np.dot(chol.T, chol)
 
     if norm_diag:
-        d = np.diag(prec)
+        # Form the diagonal vector into a row matrix
+        d = np.diag(prec).reshape(1, prec.shape[0])
         d = 1. / np.sqrt(d)
-        prec *= d
-        prec *= d[:, np.newaxis]
+
+        prec *= d * d.T
 
```

Comment:
Changed.. thanks... 

BTW could I know what this process is known as?
( I know d * d.T makes d symmetric... multiplying with prec makes the diagonals of prec all 1s... Is this known as diagonal normalization or something ( if such a word exists ) )